openedx/core/djangoapps/programs/utils.py

```diff
@@ -1028,3 +1054,51 @@ def is_user_enrolled_in_program_type(user, program_type_slug, paid_modes_only=Fa
         elif course_run_id in course_runs:
             return True
     return False
+
+
+def get_subscription_api_client(user):
+    """
+    Returns an API client which can be used to make Subscriptions API requests.
+    """
+    scopes = [
+        'user_id',
+        'email',
+        'profile'
+    ]
+    jwt = create_jwt_for_user(user, scopes=scopes)
+    client = requests.Session()
+    client.auth = SuppliedJwtAuth(jwt)
+
+    return client
+
+
+def get_programs_subscription_data(user, program_uuid=None):
+    """
+    Returns the subscription data for a user's program if uuid is specified
+    else return data for user's all subscriptions.
+    """
+    client = get_subscription_api_client(user)
+    api_path = f"{settings.SUBSCRIPTIONS_API_PATH}"
+    subscription_data = []
+
+    log.info(f"B2C_SUBSCRIPTIONS: Requesting Program subscription data for user: {user}" +
+             (f" for program_uuid: {program_uuid}" if program_uuid is not None else ""))
```

Comment:
Nit: This can be written without string concatenation doing something like...
`log.info(
    f"B2C_SUBSCRIPTIONS: Requesting Program subscription data for user: {user} "
    f" for program_uuid: {program_uuid}" if program_uuid is not None else "")
)`
Should this log be inside the try/if? That way we only log it if we indeed have a `program_uuid`? This takes out the need for a ternary expression trying to determine if we have a program UUID.

Reply:
This is a generic util function which works both for listing as well as details page. We will have both cases where we are fetching all programs(no uuid) and a single program with uuid. Without concatenation we don't get the resultant string that we need for both the cases.